sys/net/gnrc/link_layer/lwmac/tx_state_machine.c

```diff
@@ -0,0 +1,457 @@
+
+            /* Prepare packet with LwMAC header*/
+            lwmac_frame_broadcast_t hdr = {};
+            hdr.header.type = FRAMETYPE_BROADCAST;
+            hdr.seq_nr = lwmac->tx.bcast_seqnr++;
+
+            pkt->next = gnrc_pktbuf_add(pkt->next, &hdr, sizeof(hdr), GNRC_NETTYPE_LWMAC);
+
+            /* No Auto-ACK for broadcast packets */
+            netopt_enable_t autoack = NETOPT_DISABLE;
+			lwmac->netdev2_driver->set(lwmac->netdev->dev, NETOPT_AUTOACK, &autoack, sizeof(autoack));
+
+            first = true;
+        }
+
+        if( lwmac_timeout_is_expired(lwmac, TIMEOUT_NEXT_BROADCAST) ||
+            first ) {
+            /* Don't let the packet be released yet, we want to send it again */
+            gnrc_pktbuf_hold(pkt, 1);
+
+			lwmac->netdev->send(lwmac->netdev, pkt);
+            _set_netdev_state(lwmac, NETOPT_STATE_TX);
+
+            lwmac_set_timeout(lwmac, TIMEOUT_NEXT_BROADCAST, LWMAC_TIME_BETWEEN_BROADCAST_US);
+            LOG_INFO("Broadcast sent\n");
+        }
+
+        break;
+    }
+    case TX_STATE_SEND_WR:
+    {
+        LOG_DEBUG("TX_STATE_SEND_WR\n");
+
+        gnrc_pktsnip_t* pkt;
+        gnrc_netif_hdr_t *nethdr;
+        uint8_t* dst_addr = NULL;
+        int addr_len;
+
+        /* Get destination address */
+        addr_len = _get_dest_address(lwmac->tx.packet, &dst_addr);
+        if(addr_len <= 0 || addr_len > 8) {
+            LOG_ERROR("Invalid address length: %i\n", addr_len);
+            GOTO_TX_STATE(TX_STATE_FAILED, true);
+        }
+
+        /* Assemble WR */
+        lwmac_frame_wr_t wr_hdr = {};
+        wr_hdr.header.type = FRAMETYPE_WR;
+
+        pkt = gnrc_pktbuf_add(NULL, &wr_hdr, sizeof(wr_hdr), GNRC_NETTYPE_LWMAC);
+        if(pkt == NULL) {
+            LOG_ERROR("Cannot allocate pktbuf of type GNRC_NETTYPE_LWMAC\n");
+            GOTO_TX_STATE(TX_STATE_FAILED, true);
+        }
+
+        pkt = gnrc_pktbuf_add(pkt, NULL, sizeof(gnrc_netif_hdr_t) + addr_len, GNRC_NETTYPE_NETIF);
+        if(pkt == NULL) {
+            LOG_ERROR("Cannot allocate pktbuf of type GNRC_NETTYPE_NETIF\n");
+            GOTO_TX_STATE(TX_STATE_FAILED, true);
+        }
+
+        /* We wouldn't get here if adding the NETIF header had failed, so no
+           sanity checks needed */
+        nethdr = (gnrc_netif_hdr_t*) _gnrc_pktbuf_find(pkt, GNRC_NETTYPE_NETIF);
+
+        /* Construct NETIF header and insert address for WR packet */
+        gnrc_netif_hdr_init(nethdr, 0, addr_len);
+        gnrc_netif_hdr_set_dst_addr(nethdr, dst_addr, addr_len);
+
+        /* Disable Auto ACK */
+        netopt_enable_t autoack = NETOPT_DISABLE;
+		lwmac->netdev2_driver->set(lwmac->netdev->dev, NETOPT_AUTOACK, &autoack, sizeof(autoack));
+
+        /* Prepare WR, this will discard any frame in the transceiver that has
+         * possibly arrived in the meantime but we don't care at this point. */
+		lwmac->netdev->send(lwmac->netdev, pkt);
```

Comment:
And, also, please apply this return value check to all the places where Lw-MAC calls the ```send()``` function for sending packets. And take further measurements for tackling these situations, e.g., drop buffered packet to release memory.